cow_instrument/InstrumentTree.h

```diff
@@ -17,30 +16,38 @@ class Detector;
  */
 class InstrumentTree {
 public:
-  InstrumentTree(std::unique_ptr<const Node> &&root, size_t nDetectors);
-
-  InstrumentTree(const InstrumentTree& other);
-
-  std::unique_ptr<NodeIterator> iterator() const;
+  InstrumentTree(std::vector<Node> &&nodes, size_t nDetectors);
+  InstrumentTree(const InstrumentTree& original) = default;
 
   const Node &root() const;
 
   const Detector &getDetector(size_t detectorIndex) const;
 
   unsigned int version() const;
 
-  std::unique_ptr<const InstrumentTree> modify(const Command &command) const;
+  InstrumentTree modify(size_t node, const Command &command) const;
+  InstrumentTree modify(const Node *node, const Command &command) const;
```

Comment:
@SimonHeybrock That was my current thought on it. Although guaranteeing unique identity would be a problem. That is to say, when names are declare. The read/search operations would expect that the name would be unique. we'd have to enforce this at write time.